luigi/parameter.py

```diff
@@ -959,3 +959,28 @@ def serialize(self, x):
         :param x: the value to serialize.
         """
         return json.dumps(x)
+
+class ChoiceParameter(Parameter):
+    """Parameter that restricts options to the specified set
+    All choices must be of the same type that so that parse can
+    correctly infer type
+    Variable type is specified with var_type=...
+    """
+    def __init__(self, choices, var_type=str, *args, **kwargs):
+        super(ChoiceParameter, self).__init__(*args, **kwargs)
+        self.choices = set(var_type(choice) for choice in choices)
```

Comment:
Can you make this into an assertion? I imagine people creating new `Parameter`s know enough luigi+python that they would rather be explicit in types over than luigi magically converting stuff behind their back.